src/syft/lib/generic.py

```diff
@@ -16,6 +16,19 @@
 from .util import replace_classes_in_module
 
 
+class CWrapper:
```

Comment:
need more documentation on this class - please fully explain how it works

Reply:
Yup I will document next, still getting it to work unfortunately, hit some weird issues with property / method pointer return types and getattr on __id on the CWrapper.